make_prg/from_msa/cluster_sequences.py

```diff
@@ -127,6 +126,69 @@ def extract_clusters(
     return result
 
 
+class ClusteringResult(object):
+    """
+    Stores the result of clustering as a ClusteredIDs object.
+    If clustering was not meaningful (e.g. one sequence per cluster),
+    a set of sequences can be directly used as set of alternative
+    alleles of the variant site under construction.
+    """
+    def __init__(
+        self,
+        clustered_ids: ClusteredIDs,
+        sequences: Optional[Sequences] = None,
+    ):
+        self.clustered_ids: ClusteredIDs = clustered_ids
+        self.sequences: Optional[Sequences] = sequences
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self.__dict__ == other.__dict__
+        else:
+            return False
+
+    @property
+    def no_clustering(self):
+        return len(self.clustered_ids) == 1
+
+    @property
+    def have_precomputed_sequences(self):
+        return self.sequences is not None
+
+    def __repr__(self):
+        return f"ClusteringResult(clustered_ids={self.clustered_ids}, sequences={self.sequences})"
+
+    def __str__(self):
+        return self.__repr__()
+
+
+def merge_sequences(*seqlists: Sequences, first_seq: str) -> Sequences:
+    first_seq_found = False
+    result = list()
+    for seqlist in seqlists:
+        for sequence in seqlist:
+            if sequence == first_seq:
+                first_seq_found = True
+            else:
+                result.append(sequence)
+
+    # Note: this is an assert as is it the dev's responsibility to pass the correct first_seq argument (i.e. the user
+    # can't cause this error by erroneous input)
```

Comment:
I agree with using an assert instead of throwing error, and with the reasoning, however we could consider not writing that comment, and have this behaviour be a general convention project-wide. But then, for a new dev on this project, we would need a way to tell them this is what we do, and not sure how to do that :')
feel free to ignore this

Reply:
I do agree, and thought about removing the comment also! And I have this comment duplicated at least 3 times in the code base now. I think it might be better to remove the comment, and create a Coding convention wiki page? As you said, this is just a convention, nothing related to the business logic, and it end up polluting the code...

Reply:
it might be better a `coding_convention.md` file in the repo root actually

Reply:
@mbhall88 is good with best practices/conventions so i'd like to know his opinion

otherwise i would vote for putting it in wiki page

i wonder if we should have a wiki page or something for the iqbal-group organisation as this is kind of our target audience here

Reply:
but actually there is no make_prg wiki page right now and perhaps a `coding_convention.md` makes sense.

Reply:
I've removed this comment and added a `coding_convention.md` in the project's root, please check

Reply:
The convention would be to add this in a `CONTRIBUTING.md` file where we say any assert statement is "<insert the text of that comment>".
I think it's important that if there are any assert statements, they should have a detail description explaining *why* we think this particular event should not be able to happen.

Reply:
Great, done as proposed!